test/e2e/framework/node/init/init.go

```diff
@@ -30,14 +31,10 @@ import (
 func init() {
 	framework.NewFrameworkExtensions = append(framework.NewFrameworkExtensions,
 		func(f *framework.Framework) {
-			ginkgo.AfterEach(func(ctx context.Context) {
-				if f.ClientSet == nil {
-					// Test didn't reach f.BeforeEach, most
-					// likely because the test got
-					// skipped. Nothing to check...
-					return
-				}
-				e2enode.AllNodesReady(ctx, f.ClientSet, 7*time.Minute)
+			ginkgo.BeforeEach(func(ctx context.Context) {
+				ginkgo.DeferCleanup(func(ctx context.Context) {
```

Comment:
ok, this is the same as in 

https://github.com/kubernetes/kubernetes/blob/64799e076097cf3c3dd12a82f9593a9bd57f042e/test/e2e/framework/framework.go#L209-L215

and easier to think about